synapse/storage/group_server.py

```diff
@@ -55,10 +55,11 @@ def get_group(self, group_id):
                 "group_id": group_id,
             },
             retcols=(
-                "name", "short_description", "long_description", "avatar_url", "is_public"
+                "name", "short_description", "long_description",
+                "avatar_url", "is_public", "join_policy",
```

Comment:
Having looked at the sytest I now realise that this gets passed straight through to the client by `/profile`. This API doesn't make much sense as the `join_policy` isn't a string, but a dict.

As you can see the way we do this with visibility is to use a separate flag: we don't specify the precise visibility settings, but simply whether the group is fully public or not. Either we should have an `is_openly_joinable` flag or return the full `join_policy` dict. Personally I'd probably match the existing stuff and return `is_openly_joinable` (or some such).

Reply:
Ok, having discussed this IRL, we're going to go for `is_openly_joinable` as we want to maintain extensibility of the group server concept.